server/accessmanager_test.go

```diff
@@ -0,0 +1,60 @@
+package server
+import (
+	"testing"
+	"github.com/smancke/guble/guble"
+	"github.com/stretchr/testify/assert"
+	"fmt"
+)
+
+type TestAccessManager struct {
+	access map[string]map[guble.Path]bool
+}
+
+func NewTestAccessManager() *TestAccessManager {
+	return &TestAccessManager{
+		access:make(map[string]map[guble.Path]bool),
+	}
+}
+
+func (tam *TestAccessManager ) allow(userId string, path guble.Path) {
+	v, ok := tam.access[userId];
+	if(!ok) {
+		v = make(map[guble.Path]bool);
+		tam.access[userId] = v;
+	}
+	v[path] = true;
+}
+
+func (tam *TestAccessManager ) AccessAllowed(accessType AccessType, userId string, path guble.Path) bool {
+	fmt.Print("AccessAllowed: ", userId, path)
+	v, ok := tam.access[userId]
+	if(ok) {
+		_ , ok = v[path];
+		fmt.Println(" : true")
+		return ok
+	}
+	fmt.Println(" : false")
+	return false;
+}
+
+func Test_AllowAllAccessManager(t *testing.T) {
+	a := assert.New(t)
+	am := AccessManager(NewAllowAllAccessManager(true))
+	a.True(am.AccessAllowed(READ, "userid", "/path"))
+
+	am = AccessManager(NewAllowAllAccessManager(false))
+	a.False(am.AccessAllowed(READ, "userid", "/path"))
+
+}
+
+// this test needs an external service running
+// TODO create mock service for testing
+func Dont_Test_RestAccessManager(t *testing.T) {
```

Comment:
I would separate this test in the positive and negative tests and use the golang
net/http/httptest/NewServer for emulating the server, e.g.:

ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    w.Write([]byte("true")
}))
defer ts.Close()